### ISSUE TITLE
Matplotlib does not work with xeus-python

### ISSUE THREAD

**Issue by SirSharpest** [jupyter-xeus/xeus-python#224]
On XEUS `0.23.9` I find that using the `@interact` decorator and sliders that they will appear fine and as expected, however when they are attached to figures they don't display properly. 

Here is a minimal example that doesn't work on my system: 

``` python
from __future__ import print_function
from ipywidgets import interact, interactive, fixed, interact_manual
import ipywidgets as widgets
import matplotlib.pyplot as plt
@interact(n=(1,7))
def plotN(n):
    vals = [i for i in range(n)]
    plt.plot(vals)
```

expected behaviour: [img](https://i.imgur.com/dF816YP.png)

actual behaviour: [img](https://i.imgur.com/kkNqJGg.png)

**Comment by martinRenou** [comment]
Thanks a lot, I'll have a look

**Comment by martinRenou** [comment]
It's just that Matplotllib does not work by default. We'll find a way to make it work soon.

**Comment by martinRenou** [comment]
So, the `inline` Matplotlib backend (the one used by default by ipykernel) is implemented in ipykernel itself: https://github.com/ipython/ipykernel/blob/master/ipykernel/pylab/backend_inline.py

Either we:
- Rely on this implementation, if ipykernel is installed. This would require some changes on our ipykernel mock (missing `config` and `events` attribute)
- Implement our own `inline` back-end
- Use `ipympl` as default back-end (this would require making ipympl work in xeus-python, which is not the case)

**Comment by martinRenou** [comment]
On idea might be to export the https://github.com/ipython/ipykernel/blob/master/ipykernel/pylab module outside of ipykernel, make it a separate package. Then we'll reuse it in ipykernel and xeus-python. That's the cleanest solution IMO.

**Comment by SylvainCorlay** [comment]
Yes, from our earlier in-person, that seems like a sensible approach.

**Comment by fschiffers** [comment]
Would be interesting to know once this is resolved. I need matplotlib for my research, but would love to have the debugging tool available.

Thanks for creating xeus-python. Amazing tool :)

**Comment by SylvainCorlay** [comment]
@fschiffers fyi you can already use matplotlib in xeus-python through the ipympl package whcih provides interactivity in the notebook.

**Comment by christasich** [comment]
The ipympl package with xeus-python doesn't work for me. When I call `%matplotlib widget`, I get `SyntaxError: invalid syntax`. Is there a workaround that I'm missing?

**Comment by martinRenou** [comment]
@christasich it is a known issue, xeus-python does not support magics yet

**Comment by SylvainCorlay** [comment]
> The ipympl package with xeus-python doesn't work for me. When I call %matplotlib widget, I get SyntaxError: invalid syntax. Is there a workaround that I'm missing?

importing ipympl should enable the backend.

**Comment by christasich** [comment]
I've imported it and I still get the syntax error.

**Comment by SylvainCorlay** [comment]
What I mean is that once youbhave imported it, you don't need the matplotlib magics.

Btw, xeus-python 0.8.1 now supports magics, but not the matplotlib one which requires some work. Too much special casing of matplotlib in IPython.

**Comment by christasich** [comment]
I updated to `xeus-python 0.8.1` and imported `ipympl 0.5.6` but when I try to plot (`matplotlib-base 3.2.2`), I still get the generic call to the location on the disk (e.g. `<matplotlib.axes._subplots.AxesSubplot object at 0x7f1960526cd0>`). No plot is shown. I'm using JupyterLab 2.1.5 on Ubuntu 18.04. I get the same result in a console or notebook.

**Comment by gxf1986** [comment]
> I updated to `xeus-python 0.8.1` and imported `ipympl 0.5.6` but when I try to plot (`matplotlib-base 3.2.2`), I still get the generic call to the location on the disk (e.g. `<matplotlib.axes._subplots.AxesSubplot object at 0x7f1960526cd0>`). No plot is shown. I'm using JupyterLab 2.1.5 on Ubuntu 18.04. I get the same result in a console or notebook.

@SylvainCorlay @martinRenou  I met the same problem. Is there any updates about the issue? or is there any temporary solution to support matplotlib.

**Comment by SylvainCorlay** [comment]
In JupyterLab, ipympl requires installing the front-end to the Jupyter interactive widgets with 

```
jupyter labextension install @jupyter-widgets/jupyterlab-manager
```

**Comment by christasich** [comment]
I do have the front-end installed, but I still only get the object location with no plot.

`@jupyter-widgets/jupyterlab-manager v2.0.0  enabled  OK`

Is there anything else I can provide to help diagnose the issue?

**Comment by bucweat** [comment]
> I do have the front-end installed, but I still only get the object location with no plot.
> 
> `@jupyter-widgets/jupyterlab-manager v2.0.0 enabled OK`
> 
> Is there anything else I can provide to help diagnose the issue?

I had the same issue...turns out I didn't have jupyterlab-manager installed when I thought I did. Fixed that and now everything works. Here are the labextensions I currently have installed:
```
(xeus-python)` charlie@CharlieMsiMba xeus-python % jupyter labextension list                                       
JupyterLab v2.1.5
Known labextensions:
   app dir: /Users/charlie/miniconda3/envs/xeus-python/share/jupyter/lab
        @jupyter-widgets/jupyterlab-manager v2.0.0  enabled  OK
        @jupyterlab/debugger v0.3.0-beta.1  enabled  OK
        jupyter-matplotlib v0.7.2  enabled  OK
        jupyterlab-datawidgets v6.2.0  enabled  OK
```
Hope that might help someone.

**Comment by gxf1986** [comment]
@SylvainCorlay , I face the same kernel crash issue as #301 ,here is the error message:

Error: signal 11:
/root/miniconda3/bin/xpython(+0x9f8c)[0x55a90a8f5f8c]
/lib/x86_64-linux-gnu/libc.so.6(+0x3ef20)[0x7f80222aaf20]
/root/miniconda3/bin/../lib/libxeus.so.1(_ZN4xeus13xcomm_manager13register_commEN3xtl19xbasic_fixed_stringIcLm55ELi5ENS1_13string_policy12silent_errorESt11char_traitsIcEEEPNS_5xcommE+0x15)[0x7f80227bae05]
/root/miniconda3/bin/../lib/libxeus-python.so.0(+0x81306)[0x7f8022856306]
/root/miniconda3/bin/../lib/libxeus-python.so.0(+0x3e260)[0x7f8022813260]
/root/miniconda3/bin/../lib/libxeus-python.so.0(+0x217bd)[0x7f80227f67bd]
/root/miniconda3/bin/../lib/libpython3.7m.so.1.0(_PyMethodDef_RawFastCallDict+0x163)[0x7f8022030c13]
/root/miniconda3/bin/../lib/libpython3.7m.so.1.0(_PyCFunction_FastCallDict+0x26)[0x7f8022031466]
/root/miniconda3/bin/../lib/libpython3.7m.so.1.0(_PyObject_Call_Prepend+0xde)[0x7f802203312e]
/root/miniconda3/bin/../lib/libpython3.7m.so.1.0(PyObject_Call+0x68)[0x7f80220316b8]

**Comment by arush1947** [comment]
Any updates on this issue? Unable to generate any plots using matplotlib. importing ipympl.  jupyterlab-manager installed and enabled.

JupyterLab v2.1.5
Known labextensions:
   app dir: \anaconda3\envs\Xeus Env\share\jupyter\lab
        @jupyter-widgets/jupyterlab-manager v2.0.0 enabled  ok
        @jupyterlab/commenting-extension v0.3.0 enabled  ok
        @jupyterlab/debugger v0.2.0 enabled  ok
        @jupyterlab/git v0.20.0 enabled  ok
        @jupyterlab/shortcutui v0.5.0-beta.2 enabled  ok
        @lckr/jupyterlab_variableinspector v0.5.1 enabled  ok
        jupyter-matplotlib v0.7.3 enabled  ok
        jupyterlab-datawidgets v6.3.0 enabled  ok
        nbdime-jupyterlab v2.0.0 enabled  ok

**Comment by pratikchhapolika** [comment]
Any update on the issue. Not able to use magic command and also matplotlib.?

**Comment by zombba** [comment]
> > The ipympl package with xeus-python doesn't work for me. When I call %matplotlib widget, I get SyntaxError: invalid syntax. Is there a workaround that I'm missing?
> 
> importing ipympl should enable the backend.

import ipympl dont enable backend
